### PR TITLE
CreateIfNotExists/DeleteIfExists logs exception

### LINKED ISSUE #9275: [storage] [blobs] Update CreateIfNotExists/DeleteIfExists to log exception thrown
Currently the CreateIfNotExists/DeleteIfExists does not log the exception when it is thrown like how it is done for Exists for blobs.

Before:
```
catch (RequestFailedException storageRequestFailedException)
when (storageRequestFailedException.ErrorCode == Constants.Blob.AlreadyExists)
{
           return default;
}
```
After:
```
 catch (RequestFailedException storageRequestFailedException)
when (storageRequestFailedException.ErrorCode == Constants.Blob.NotFound)
 {
         return Response.FromValue(false, default);
}
catch (Exception ex)
{
         Pipeline.LogException(ex);
         throw;
}
```

### PR DESCRIPTION
**Pull request by amnguye**
Resolves #9275

### CODE REVIEW

**Review by seanmcc-msft — APPROVED** [review]

**Comment by seanmcc-msft** [comment]
Do you have unit tests for when the blob exists, doesn't exist, and another exception is thrown?

**Comment by amnguye** [comment]
> Do you have unit tests for when the blob exists, doesn't exist, and another exception is thrown?

Just pushed up some more changes to add the tests missing the error/exception tests. The ones that were missing were for BlobBaseClient for DeleteIfExists and ContainerClient for CreateIfNotExists/DeleteIfNotExists. The other tests exists already.

**Review by seanmcc-msft — APPROVED** [review]

File: sdk/storage/Azure.Storage.Blobs/src/AppendBlobClient.cs [hunk 1]
```diff
@@ -410,22 +410,40 @@ public new AppendBlobClient WithSnapshot(string snapshot)
...
-            }
-            catch (RequestFailedException storageRequestFailedException)
-            when (storageRequestFailedException.ErrorCode == Constants.Blob.AlreadyExists)
+            using (Pipeline.BeginLoggingScope(nameof(AppendBlobClient)))
             {
-                return default;
+                Pipeline.LogMethodEnter(
+                    nameof(AppendBlobClient),
+                    message:
+                    $"{nameof(Uri)}: {Uri}");
```
Comment (JoshLove-msft) [inline sdk/storage/Azure.Storage.Blobs/src/AppendBlobClient.cs]: Any reason to not match the message format used in `CreateInternal`?
It also includes the `BlobHttpHeaders`

**Review by tg-msft — APPROVED** [review]
Looks good!

**Pull request merged**